### PR TITLE
Add assignment dates to discussion details

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-14386
affects: Teacher
release note: none

### CODE REVIEW

File: Core/Core/Discussions/DiscussionDetailsViewController.swift [hunk 1]
```diff
@@ -227,12 +227,17 @@ public class DiscussionDetailsViewController: UIViewController, ColoredNavViewPr
         publishedLabel.textColor = .named(isPublished ? .textSuccess : .textDark)
         publishedView.isHidden = env.app != .teacher || isAnnouncement || showRepliesToEntryID != nil
 
-        if let assignmentID = topic.first?.assignmentID {
+        if env.app == .teacher, let courseID = courseID, let assignmentID = topic.first?.assignmentID {
```
Comment (vanwagonet) [inline Core/Core/Discussions/DiscussionDetailsViewController.swift:230]: this wasn't only for teacher app before. oops.

File: Core/Core/Submissions/SubmissionBreakdownViewController.swift [hunk 2]
```diff
@@ -91,12 +91,11 @@ class SubmissionBreakdownViewController: UIViewController {
         onPaperLabel.isHidden = true
         noSubmissionsLabel.isHidden = true
         noSubmissionsLabel.text = NSLocalizedString("Tap to view submissions list.", bundle: .core, comment: "")
-
-        summary.refresh()
     }
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
+        summary.refresh(force: true)
```
Comment (vanwagonet) [inline Core/Core/Submissions/SubmissionBreakdownViewController.swift:98]: I guess I hadn't been refreshing the summary on willAppear, so I fixed that.

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/hipoyia) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fhipoyia)
teacher: [Link](https://rebrand.ly/2s4pmzp) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F2s4pmzp)
parent: [Link](https://rebrand.ly/7udtgtt) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F7udtgtt)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Teacher

[MBL-14386](https://instructure.atlassian.net/browse/MBL-14386)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.36% | 94.19% | -0.83%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/CoreTests/Assignments/AssignmentDatesViewControllerTests.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            lock_at: DateComponents(calendar: .current, year: 2030, month: 6, day: 1).date
+        ), ]))
+        controller.view.layoutIfNeeded()
+        controller.viewWillAppear(false)
+        XCTAssertEqual(getText(), """
+        Due: May 20, 2020 at 12:00 AM
+        For: Everyone
+        Available From: May 1, 2020 at 12:00 AM
+        Available Until: Jun 1, 2030 at 12:00 AM
+        """)
```
Comment (armstrongnate) [inline Core/CoreTests/Assignments/AssignmentDatesViewControllerTests.swift:72]: Beautiful!

**Pull request merged**